template.yaml

```diff
@@ -89,14 +167,231 @@ Globals:
+        Metric alarm to trigger ECS scale down
+      AlarmActions:
+        - !Ref FrontendAutoScalingPolicyScaleIn
+      ComparisonOperator: LessThanThreshold
+      EvaluationPeriods: 5
+      MetricName: CPUUtilization
+      Namespace: AWS/ECS
+      Period: 60
+      Statistic: Average
+      Threshold: 50
+      DatapointsToAlarm: 5
+      Dimensions:
+        - Name: ClusterName
+          Value: !Ref FrontendECSCluster
+        - Name: ServiceName
+          Value: !GetAtt FrontendECSService.Name
+
+  FrontendAutoScalingTarget:
+    Type: AWS::ApplicationAutoScaling::ScalableTarget
+    Properties:
+      MaxCapacity: !FindInMap [ EnvironmentConfiguration, !Ref Environment, frontendAutoScalingMaxCount ]
+      MinCapacity: !FindInMap [ EnvironmentConfiguration, !Ref Environment, frontendAutoScalingMinCount ]
+      ResourceId: !Join
+        - '/'
+        - - "service"
+          - !Ref FrontendECSCluster
+          - !GetAtt FrontendECSService.Name
+      RoleARN: !Sub "arn:aws:iam::${AWS::AccountId}:role/aws-service-role/ecs.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_ECSService"
+      ScalableDimension: ecs:service:DesiredCount
+      ServiceNamespace: ecs
+
+  FrontendAutoScalingPolicyScaleOut:
+    DependsOn: FrontendAutoScalingTarget
+    Type: AWS::ApplicationAutoScaling::ScalingPolicy
+    Properties:
+      PolicyName: !Sub "${AWS::StackName}-FrontendAutoScalingPolicyScaleOut"
+      PolicyType: StepScaling
+      ResourceId: !Join
+        - '/'
+        - - "service"
+          - !Ref FrontendECSCluster
+          - !GetAtt FrontendECSService.Name
+      ScalableDimension: ecs:service:DesiredCount
+      ServiceNamespace: ecs
+      StepScalingPolicyConfiguration:
+        AdjustmentType: PercentChangeInCapacity
+        Cooldown: 60
+        MetricAggregationType: Average
+        MinAdjustmentMagnitude: 10
+        StepAdjustments:
+          - ScalingAdjustment: 300
+            MetricIntervalLowerBound: 0
+            MetricIntervalUpperBound: 10
+          - ScalingAdjustment: 600
+            MetricIntervalLowerBound: 10
+            MetricIntervalUpperBound: 30
+          - ScalingAdjustment: 1000
+            MetricIntervalLowerBound: 30
+
+  FrontendAutoScalingPolicyScaleIn:
+    DependsOn: FrontendAutoScalingTarget
+    Type: AWS::ApplicationAutoScaling::ScalingPolicy
+    Properties:
+      PolicyName: !Sub "${AWS::StackName}-FrontendAutoScalingPolicyScaleIn"
+      PolicyType: StepScaling
+      ResourceId: !Join
+        - '/'
+        - - "service"
+          - !Ref FrontendECSCluster
+          - !GetAtt FrontendECSService.Name
+      ScalableDimension: ecs:service:DesiredCount
+      ServiceNamespace: ecs
+      StepScalingPolicyConfiguration:
+        AdjustmentType: PercentChangeInCapacity
+        Cooldown: 240
+        MetricAggregationType: Average
+        MinAdjustmentMagnitude: 5
+        StepAdjustments:
+          - ScalingAdjustment: -50
```

Comment:
There is new Adjustment recently pushed for Scale In policy  & Scale In Alarm if you want to use that

Reply:
Applied